types/utils.d.ts

```diff
@@ -18,6 +20,19 @@ export type ObjectOrFunc<TResult, TArg = {}> = ((arg: TArg) => TResult) | TResul
 // Props
 // ========================================================
 
-export type ItemShorthand = React.ReactNode | object
+export type IProps = ObjectOf<any>
 export type ReactChildren = React.ReactNodeArray | React.ReactNode
 export type ComponentEventHandler<TProps> = (event: React.SyntheticEvent, data: TProps) => void
+
+// ========================================================
+// Shorthand Factories
+// ========================================================
+
+export type ShorthandPrimitive = string | number
+export type ShorthandValue = React.ReactNode | IProps
+export type MapValueToProps = (value: ShorthandPrimitive) => IProps
+export type ShorthandRenderFunction = (
+  Component: React.ReactType,
+  props: IProps,
+  children: ReactChildren,
+) => React.ReactElement<any>
```

Comment:
SUIR now has following typings:

```
export type SemanticShorthandItemFunc<TProps> = (
  component: React.ComponentType<TProps>,
  props: TProps,
  children?: React.ReactChildren,
) => React.ReactElement<any> | null
```

This allows to access passed props.

Reply:
Nice.  We were just discussing improving typings for shorthand in Stardust as well.

Note on SUIR's typings here:
- `React.ComponentType` only includes Component and SFC but not strings (`'div'`, etc).  Use `React.ReactType` to include all valid component types.
- `React.ReactChildren` is actually the API for the `React.Children` utility.  Use `React.ReactNodeArray | React.ReactNode` for return values from render functions.